omp/CMakeLists.txt

```diff
@@ -15,17 +15,20 @@ set(SOURCES
     stop/criterion_kernels.cpp
     stop/residual_norm_reduction_kernels.cpp)
 
-find_package(OpenMP)
-if(OPENMP_FOUND)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
-else()
-    set(BUILD_OMP OFF CACHE BOOL "Compile OpenMP kernels for CPU" FORCE)
-endif()
+find_package(OpenMP REQUIRED)
 
 add_library(ginkgo_omp
     $<TARGET_OBJECTS:ginkgo_omp_device>
     ${SOURCES})
-target_link_libraries(ginkgo_omp PUBLIC ginkgo_cuda)
+
+target_link_libraries(ginkgo_omp PRIVATE "${OpenMP_CXX_FLAGS}")
```

Comment:
Question to which I could not find an answer on the web, does it work if you use `OpenMP_CXX_LIBRARIES` instead here? That sound like the correct thing to use.

Reply:
I increased the required version of CMake now to 3.9 and added proper linking with `OpenMP_CXX_LIBRARIES`